src/python/pants/engine/legacy/structs.py

```diff
@@ -392,32 +385,29 @@ def to_path_globs(self, relpath: str, conjunction: GlobExpansionConjunction) ->
       conjunction=conjunction,
     )
 
-  def _gen_init_args_str(self):
+  def _gen_init_args_str(self) -> str:
     all_arg_strs = []
-    positional_args = ', '.join([repr(p) for p in self._patterns])
+    positional_args = ', '.join(repr(p) for p in self._patterns)
     if positional_args:
       all_arg_strs.append(positional_args)
-    keyword_args = ', '.join([
-      '{}={}'.format(k, repr(v)) for k, v in self._kwargs.items()
-    ])
-    if keyword_args:
-      all_arg_strs.append(keyword_args)
-
+    all_arg_strs.append(f"spec_path={self._spec_path}")
+    if self._raw_exclude:
+      all_arg_strs.append(f"exclude={self._raw_exclude}")
```

Comment:
`self._kwargs` only ever had `spec_path` and `exclude`. Now, we handle these cases explicitly.